components/Common/ShellBox/index.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import styles from './index.module.scss';
+import ReactDomServer from 'react-dom/server';
+
+interface Props {
+  children: React.ReactNode;
+  textToCopy?: string;
+}
+
+const ShellBox = ({ children, textToCopy }: React.PropsWithChildren<Props>) => {
+  const [copied, copyText] = useCopyToClipboard();
+
+  const handleCopyCode = async (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    const text = textToCopy? textToCopy : ReactDomServer.renderToString(children);
```

Comment:
I think this could be an `useCallback`, as we don't want to recalculate the textToCopy or the children with `ReactDomServer` on each click.